authentication/auth.go

```diff
@@ -158,6 +207,77 @@ func generateClaims(claimValues map[string][]string, launcherSchema surveys.Laun
 	return claims
 }
 
+func generateClaimsV2(claimValues map[string][]string, launcherSchema surveys.LauncherSchema) (claims map[string]interface{}) {
+
+	var roles []string
+	if rolesValues, ok := claimValues["roles"]; ok {
+		roles = rolesValues
+	} else {
+		roles = []string{"dumper"}
+	}
+
+	claims = make(map[string]interface{})
+
+	claims["roles"] = roles
+	TxID, _ := uuid.NewV4()
+	claims["tx_id"] = TxID.String()
+	claims["version"] = "v2"
+
+	surveyMetadata := make(map[string]interface{})
+	data := make(map[string]interface{})
+
+	if launcherSchema.SurveyType == "social" {
+		receiptingKeys := []string{"questionnaire_id"}
+		surveyMetadata["receipting_keys"] = receiptingKeys
+
+		for key, value := range claimValues {
+			if isSurveyMetadataSocial(key) == true {
+				data[key] = value[0]
+			} else {
+				if key != "roles" {
+					if isRequiredMetadata(key) {
+						if value[0] != "" {
+							claims[key] = value[0]
+						}
+					}
+				}
+				if key == "roles" {
+					claims[key] = value
+				}
+			}
+			surveyMetadata["data"] = data
+			claims["survey_metadata"] = surveyMetadata
+		}
+	} else {
+		for key, value := range claimValues {
+			if isSurveyMetadataBusiness(key) == true {
+				data[key] = value[0]
+			} else {
+				if key != "roles" {
+					if isRequiredMetadata(key) {
+						if value[0] != "" {
+							claims[key] = value[0]
+						}
+					}
+				} else {
+					claims[key] = value
+				}
+			}
+		}
+		surveyMetadata["data"] = data
+		claims["survey_metadata"] = surveyMetadata
+	}
```

Comment:
This is quite difficult to digest due to the indentation. I would break this up. It would be nice to simple have a `getSurveyMetadataFromClaims` function.

- Why do we need both `isSurveyMetadataSocial` and `isSurveyMetadataBusiness`?. We just need to know if it is survey metadata?. The UI shouldn't allow posting business claims if the schema was social. Only reason we would need to know if it is social vs business is if the same key had different uses?. Let me know if I misunderstood.

Reply:
I've refactored this now, I was able to removed both methods now I updated the js to stop ru_ref from being passed.